<!-- Launch file for naive_motion_predict -->
<launch>
    <!-- 预测点间隔0.1s，总预测点数80，即预测未来8s的轨迹-->
    <arg name="interval_sec" default="0.1"/>
    <arg name="num_prediction" default="80"/>
    <arg name="sensor_height" default="0"/>
    <arg name="filter_out_close_object_threshold" default="1.5"/>
    <arg name="input_topic" default="/detection/objects"/>

    <!-- 0通过算法预测 1通过记录的轨迹预测 -->
    <arg name="predictionByrecorded_trajectories" default="false"/>
    <arg name="waypoints_file" default="$(find naive_motion_predict)/scripts/files"/>

    <node pkg="naive_motion_predict" type="naive_motion_predict" name="naive_motion_predict" output="screen">
        <param name="interval_sec" value="$(arg interval_sec)"/>
        <param name="num_prediction" value="$(arg num_prediction)"/>
        <param name="sensor_height" value="$(arg sensor_height)"/>
        <param name="filter_out_close_object_threshold" value="$(arg filter_out_close_object_threshold)"/>
        <param name="predictionByrecorded_trajectories" value="$(arg predictionByrecorded_trajectories)"/>
        <param name="waypoints_file" value="$(arg waypoints_file)" />


        <!-- 启用基于中心线的预测 -->
        <param name="predictionByCenterline" value="false"/>

        <!-- 启用多模态预测 -->
        <param name="predictionMultiModal" value="true"/>
        <param name="lane_change_probability" value="0.3"/>
        <param name="lane_follow_probability" value="0.7"/>
        
        
        <!-- 中心线文件列表 -->
        <rosparam param="centerline_files">
          ["/home/xining095/Carmaker/smartcar/miniEV_grad_xn/src/map/tracks/route_0longspline.txt", 
          "/home/xining095/Carmaker/smartcar/miniEV_grad_xn/src/map/tracks/route_1longspline.txt", 
          "/home/xining095/Carmaker/smartcar/miniEV_grad_xn/src/map/tracks/route_2longspline.txt"]

        </rosparam>
    </node>



    <!-- <remap from="/ITS/DetectObjects" to="$(arg input_topic)"/> -->

    <!-- <node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="naive_prediction_visualization_01"
          output="screen" ns="/prediction/motion_predictor" /> -->
</launch>
